src/utils.js

```diff
@@ -64,8 +64,39 @@ const utils = {
     hasSingleItemOf(arr, itemValue) {
         return arr.length === 1
             && (arguments.length === 2 ? arr[0] === itemValue : true);
-    }
+    },
+
+    isArrIndex (note) {
+        return /^\[\d+\]$/.test(note);
+    },
+
+    splitNotation (notation) {
+        // needs rethink
```

Comment:
It doesn't support nested array accessors `prop[0][1]`.
I'll try to update it tomorrow.